clients/src/main/java/org/apache/kafka/clients/producer/ProducerRecord.java

```diff
@@ -56,6 +59,8 @@
     public ProducerRecord(String topic, Integer partition, Long timestamp, K key, V value) {
         if (topic == null)
             throw new IllegalArgumentException("Topic cannot be null");
+        if (!TopicValidator.isValidateTopicName(topic))
+            throw new InvalidTopicException(topic + " is not a valid topic name.");
```

Comment:
That seems like a good suggestion as core includes clients as a dependency.  @becketqin included a comment in the JIRA ticket that had a different preference.  @granthenke can you chime in on that?